### PR TITLE
Dev-Francois

### PR DESCRIPTION
**Pull request by sangfrois**
# Description

I've made small changes, but also ambitious changes.

**Small** : 
- Typos in my name in manuscript, readme, etc. **My name had a typo on my own GitHub profile** 🥇 👍 
- I've added some testing for `rsp_rrv()`

**Ambitious** : 
- I've **added a README** in `/docs/examples` so that we could thorougly describe example cases and highlight contributors work. 

- In `tree/README` and `docs/examples/README`,  you'll see i added a **MyBinder** link so that *people without a conda env could run through examples and play around with parameters*. Refer to `.binder` in root for the `requirements.txt` - add yours if you're using it in your example and it's missing.

- Added an example on cardiorespiratory pattern extraction (to be continued, with HRV measures and PCA)

- In the same vein, note I added a feature to `epochs_create()` - it is illustrated in the mentionned example `cardiorespiratory.ipynb`

# Proposed Changes

I added a boolean option (`phys_event`, defaults to False) to specify that `events` argument is based on physiological activity. That way, [`epoch_create()`](https://github.com/sangfrois/NeuroKit/blob/dev/neurokit2/epochs/epochs_create.py) tolerates variable length duration of event instead of staying 1 sec. All of my argument is in [`cardiorespiratory.ipynb`](https://github.com/sangfrois/NeuroKit/blob/dev/docs/examples/cardiorespiratory.ipynb).

Follow the links to my fork

### CODE REVIEW

**Comment by DominiqueMakowski** [comment]
Awesome, thanks!! Let me give it a look ☺️

File: neurokit2/epochs/epochs_create.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 from ..misc import listify
 
 
-def epochs_create(data, events, sampling_rate=1000, epochs_start=0, epochs_end=1, event_labels=None, event_conditions=None, baseline_correction=False):
+
+def epochs_create(data, events, sampling_rate=1000, epochs_start=0, epochs_end=1, phys_event=False, event_labels=None, event_conditions=None, baseline_correction=False):
```
Comment (DominiqueMakowski) [inline neurokit2/epochs/epochs_create.py]: I am somehow not sure about this argument `phys_event` as it seems a bit too specific. Here's an alternative:

Currently, I think `epochs_start` and `epochs_end` accept lists, allowing for variable epochs lengths (although we should mention this possibility in the docstring). Thus, it's better if the user explicitly derives the end for each event (`end = np.insert(start[1:] - start[:-1], 0, start[0]); end = end/sampling_rate`) and then passes it to `epochs_end`. I think it's clearer and not really much more complicated. (and it forces an important step to be made explicit, which is a good thing IMO).

File: docs/examples/rrv.ipynb [hunk 2]
```diff
@@ -4,22 +4,19 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Respiratory Rate Variability (RRV)"
+    "# Respiratory Rate Variability Analysis"
```
Comment (DominiqueMakowski) [inline docs/examples/rrv.ipynb]:
```suggestion
    "# Respiratory Rate Variability (RRV)"
```

It's more straightforward (the "analysis" word is unnecessary here)

File: docs/examples/rrv.ipynb [hunk 3]
```diff
@@ -4,22 +4,19 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Respiratory Rate Variability (RRV), or variations in respiratory rhythm, are crucial indices of general health and respiratory complications. This example shows how to use NeuroKit to perform RRV analysis.\n",
-    "\n",
-    "\n",
-    "**I don't have any respiration data :'(**: No problem, see how you can extract **respiration from cardiac activity** (e.g., ECG) in [this example](https://github.com/neuropsychology/NeuroKit/issues/87)  [TODO]"
```
Comment (DominiqueMakowski) [inline docs/examples/rrv.ipynb]: You removed this added line (maybe you didn't update the latest version from dev)

File: docs/examples/rrv.ipynb [hunk 4]
```diff
@@ -68,17 +66,16 @@
...
+    "data = pd.read_csv(\"https://raw.githubusercontent.com/neuropsychology/NeuroKit/master/data/example_bio_100hz.csv\")\n",
+    "rsp = data[\"RSP\"]\n",
+    "nk.signal_plot(rsp) # Visualize"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Now that we have the data, we can run the processing routine using `rsp_process()` that will compute the respiration rate and extract the inhalation peaks. The function outputs two things, a dataframe and a dictionnary."
```
Comment (DominiqueMakowski) [inline docs/examples/rrv.ipynb]: Yeah I think you didn't update the latest version in which I simplified this example. Could you restore it before we merge (https://github.com/neuropsychology/NeuroKit/blob/dev/docs/examples/rrv.ipynb)

File: README.rst [hunk 5]
```diff
@@ -209,16 +213,19 @@ You can run:
 
     You can cite NeuroKit as follows:
 
-    - Makowski, D., Pham, T., Lau, Z. J., Brammer, J. C., Pham, H., Lesspinasse, F., Schölzel, C., & S H Chen, A. (2020). NeuroKit2: A Python Toolbox for Neurophysiological Signal Processing. Retrieved March 10, 2020, from https://github.com/neuropsychology/NeuroKit
+
+
+    - Makowski, D., Pham, T., Lau, Z. J., Brammer, J. C., Pham, H., Lespinasse, F., & S H Chen, A. (2020). NeuroKit: The Python Toolbox for Neurophysiological Signal Processing. Retrieved March 10, 2020, from https://github.com/neuropsychology/NeuroKit
```
[the remark this answers is not in the document]
Comment (sangfrois) [reply README.rst]: 👍

File: docs/examples/eventrelated.ipynb [hunk 6]
```diff
@@ -84,16 +101,16 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The data was obtained using 4 channels, namely **EDA**, **ECG**, **RSP** and the **Photosensor**. The onsets of events are marked in the photosensor's signal.\n",
-    "\n",
     "These events can be localized and extracted using [events_find()](https://neurokit2.readthedocs.io/en/latest/functions.html#neurokit2.events_find). \n",
     "\n",
-    "*Note that you should also specify whether to select events that are higher or below the threshold using the `threshold_keep` argument.*"
+    "*Note: that you should also specify whether to select events that are higher or below the threshold using the `threshold_keep` argument.*"
```
[the remark this answers is not in the document]
Comment (sangfrois) [reply docs/examples/eventrelated.ipynb]: hehe, you're right! Typos seem to be my thing

File: neurokit2/epochs/epochs_create.py (hunk 1 above)
Comment (sangfrois) [reply neurokit2/epochs/epochs_create.py]: You wrote : *I think `epochs_start` and `epochs_end` accept lists* - so I'll give it a try. Do some testing. 
But, I get what you are saying : functions have to stay simple and elegant.

File: docs/examples/rrv.ipynb (hunk 3 above)
Comment (sangfrois) [reply docs/examples/rrv.ipynb]: Still working on my Git workflow. Sorry for that

File: docs/examples/eventrelated.ipynb [hunk 7]
```diff
@@ -511,8 +616,20 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "As we can see, there seems to be a difference between the negative and the neutral pictures. Negative stimuli, as compared to neutral stimuli, were related to a stronger cardiac deceleration (i.e., higher heart rate variability), an accelerated breathing rate, and higher SCR magnitude."
+    "*Then interpret* : As we can see, there seems to be a difference between the negative and the neutral pictures. Negative stimuli, as compared to neutral stimuli, were related to a stronger cardiac deceleration (i.e., higher heart rate variability), an accelerated breathing rate, and higher SCR magnitude.\n",
+    "\n",
+    "## Important remarks:\n",
+    "We hope it has been helpful for your own analyses. You can't break anything if you're on Binder, so have fun. Keep in mind that *this is for illustration purposes only*. Data size limits on Github force us to downsample and have only one participant (but, it's free so...). \n",
```
[the remark this answers is not in the document]
Comment (sangfrois) [reply docs/examples/eventrelated.ipynb]: Cool!
But I kept the "disclaimer" part about sample rate and limited number of observations. 
That sounds ok ? [`eventrelated.ipynb`](https://github.com/sangfrois/NeuroKit/blob/dev/docs/examples/eventrelated.ipynb)

File: neurokit2/epochs/epochs_create.py (hunk 1 above)
Comment (sangfrois) [reply neurokit2/epochs/epochs_create.py]: If we can do like you say without any problem, I'll just revert back to the last one commited in `neuropsychology/dev` and examplify in cardioresp notebook :)

File: docs/examples/rrv.ipynb (hunk 4 above)
Comment (sangfrois) [reply docs/examples/rrv.ipynb]: Done! Thanks for reviewing and commenting everything, really nice of you!

**Comment by sangfrois** [comment]
Oh no, what did I do!? This happened too quickly for me. Sorry. I wanted to merge without my proposition in `epochs_create`

File: neurokit2/epochs/epochs_create.py (hunk 1 above)
Comment (sangfrois) [reply neurokit2/epochs/epochs_create.py]: Amidst, all the back and forth with PRs, I forgot to mention that I had tried to derive variable length epochs by giving a list of the difference between successive onset. I had this error a couple of times :
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

I'll let you deliberate on the usefulness of my add-on, maybe let you try for yourself if you have time. In the mean time, I'll just remove my example from the PR.

Comment (sangfrois) [reply neurokit2/epochs/epochs_create.py]: ``site-packages\neurokit2\epochs\epochs_create.py in epochs_create(data, events, sampling_rate, epochs_start, epochs_end, phys_event, event_labels, event_conditions, baseline_correction)
``

```         #Find the maximum numbers in an epoch

--> 104         epoch_max_duration = int(max((i * sampling_rate for i in parameters["duration"])))
    105 
    106         # Then extend data by the max samples in epochs * NaN```

**Pull request merged**